waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java

```diff
@@ -656,19 +660,61 @@ public void metastoreTunnelConfiguration() throws Exception {
     runner = WaggleDanceRunner
         .builder(configLocation)
         .primary("primary", localServer.getThriftConnectionUri(), READ_ONLY)
-        .federateWithMetastoreTunnel("waggle_remote", remoteServer.getThriftConnectionUri(), REMOTE_DATABASE, route,
-            privateKeys, knownHosts)
+        .federateWithMetastoreTunnel(SECONDARY_METASTORE_NAME, remoteServer.getThriftConnectionUri(), REMOTE_DATABASE,
+            route, privateKeys, knownHosts)
         .build();
 
     runWaggleDance(runner);
     RestTemplate rest = new RestTemplateBuilder().build();
     FederatedMetaStore federatedMetastore = rest
-        .getForObject("http://localhost:18000/api/admin/federations/waggle_remote", FederatedMetaStore.class);
+        .getForObject("http://localhost:18000/api/admin/federations/" + SECONDARY_METASTORE_NAME,
+            FederatedMetaStore.class);
 
     MetastoreTunnel metastoreTunnel = federatedMetastore.getMetastoreTunnel();
     assertThat(metastoreTunnel.getRoute(), is(route));
     assertThat(metastoreTunnel.getKnownHosts(), is(knownHosts));
     assertThat(metastoreTunnel.getPrivateKeys(), is(privateKeys));
   }
 
+  @Test
+  public void getDatabaseFromPatternManual() throws Exception {
+    runner = WaggleDanceRunner
+        .builder(configLocation)
+        .databaseResolution(DatabaseResolution.MANUAL)
+        .overwriteConfigOnShutdown(false)
+        .primary("primary", localServer.getThriftConnectionUri(),
+            AccessControlType.READ_AND_WRITE_AND_CREATE_ON_DATABASE_WHITELIST)
+        .federate(SECONDARY_METASTORE_NAME, remoteServer.getThriftConnectionUri(), "remote.?database")
+        .build();
+    runWaggleDance(runner);
+
+    HiveMetaStoreClient proxy = getWaggleDanceClient();
+    List<String> allDatabases = proxy.getAllDatabases();
+
+    assertTrue(allDatabases.contains(REMOTE_DATABASE));
+
+    proxy.close();
```

Comment:
why are you closing this, does it need to be. We don't do it anywhere else, probably ok.
If we need to close this we need to do it differently because this close won't be called when the test throws an exception etc...